### PR TITLE
samples/fs/fat_fs: Support for nrf52840dk_nrf52840 board

### PR DESCRIPTION
**Pull request by de-nordic**
PR consists of commits that add support for nrf52840dk_nrf52840 board with FAT FS partition created on internal SoC flash and QSPI connected MX25 device.

### CODE REVIEW

File: samples/subsys/fs/fat_fs/Kconfig [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+mainmenu "FAT Filesystem Sample Application"
+
+config SAMPLE_FATFS_CREATE_SOME_ENTRIES
```
Comment (danieldegrasse) [inline samples/subsys/fs/fat_fs/Kconfig:9]: Why not make this Kconfig `default y`, or even hardcode the sample to create files when none exist? I think this functionality is useful beyond just QSPI, I often test the fat_fs sample with a freshly formatted SD card, and creating a file would be useful behavior.

File: samples/subsys/fs/fat_fs/README.rst [hunk 2]
```diff
@@ -7,27 +7,64 @@ Overview
...
-************
+To access device the sample uses :ref:`disk_access_api`.
+
+Requirements for SD card support
+********************************
 
 This project requires SD card support and microSD card formatted with FAT filesystem.
 See the :ref:`disk_access_api` documentation for Zephyr implementation details.
+Boards that by default use SD card for storage:
+``arduino_mkrzero``, ``esp_wrover_kit``, ``mimxrt1050_evk``, ``nrf52840_blip``,
```
Comment (danieldegrasse) [inline samples/subsys/fs/fat_fs/README.rst:20]: mimrt1050 is in this list twice. I would also add the `mimxrt1060_evk`, `lpcxpresso55s69_cpu0`, `mimxrt1170_evk_cm7`, and `mimxrt1024_evk`. (I believe any board with the `zephyr,sdmmc-disk` compatible in its devicetree should use the SD card by default)

File: samples/subsys/fs/fat_fs/Kconfig (hunk 1 above)
Comment (de-nordic) [reply samples/subsys/fs/fat_fs/Kconfig:9]: I am fine with that.

File: samples/subsys/fs/fat_fs/README.rst (hunk 2 above)
Comment (de-nordic) [reply samples/subsys/fs/fat_fs/README.rst:20]: Will fix.

**Review by danieldegrasse — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**